go/obscuronode/host/host.go

```diff
@@ -520,21 +520,27 @@ func (a *Node) checkForSharedSecretRequests(block obscurocommon.EncodedBlock) {
 func (a *Node) monitorBlocks() {
 	listener := a.ethClient.BlockListener()
 	nodecommon.LogWithID(a.shortID, "Start monitoring Ethereum blocks..")
-	for {
-		latestBlkHeader := <-listener
-		block, err := a.ethClient.FetchBlock(latestBlkHeader.Hash())
-		if err != nil {
-			log.Panic("could not fetch block for hash %s. Cause: %s", latestBlkHeader.Hash().String(), err)
-		}
-		blockParent, err := a.ethClient.FetchBlock(block.ParentHash())
-		if err != nil {
-			log.Panic("could not fetch block's parent with hash %s. Cause: %s", block.ParentHash().String(), err)
-		}
 
-		nodecommon.LogWithID(a.shortID, "Received a new block b_%d(%d)",
-			obscurocommon.ShortHash(latestBlkHeader.Hash()),
-			latestBlkHeader.Number.Uint64())
-		a.RPCNewHead(obscurocommon.EncodeBlock(block), obscurocommon.EncodeBlock(blockParent))
+	for atomic.LoadInt32(a.interrupt) == 0 {
+		select {
+		case latestBlkHeader := <-listener:
+			block, err := a.ethClient.FetchBlock(latestBlkHeader.Hash())
+			if err != nil {
+				log.Panic("could not fetch block for hash %s. Cause: %s", latestBlkHeader.Hash().String(), err)
+			}
+			blockParent, err := a.ethClient.FetchBlock(block.ParentHash())
+			if err != nil {
+				log.Panic("could not fetch block's parent with hash %s. Cause: %s", block.ParentHash().String(), err)
+			}
+
+			nodecommon.LogWithID(a.shortID, "Received a new block b_%d(%d)",
+				obscurocommon.ShortHash(latestBlkHeader.Hash()),
+				latestBlkHeader.Number.Uint64())
+			a.RPCNewHead(obscurocommon.EncodeBlock(block), obscurocommon.EncodeBlock(blockParent))
+
+		case <-time.After(1 * time.Second):
+			// check for interrupt
```

Comment:
@BedrockSquirrel Can we explain this better? I guess you're saying we only want to attempt to fetch new blocks every second?